### ISSUE TITLE
References to private repository github.com/stackrox/rox ?

### ISSUE THREAD

**Issue by 02fa** [stackrox/stackrox#6255]
https://github.com/stackrox/stackrox/blob/bf6ceee9e907fb59d00b3c606004745a1c985947/go.mod#LL1C31-L1C31

`github.com/stackrox/rox` isn't Open Source. 

There seem to be a lot of similar references to closed source packages on this repository:

https://github.com/stackrox/stackrox/search?q=stackrox/rox

I have to confess Go isn't my cup of tea . Home comes OSS project points at a private repository? Or am I barking up the wrong tree?

**Comment by msugakov** [comment]
> `github.com/stackrox/rox` isn't Open Source.

It wasn't Open Source before it got copied to `github.com/stackrox/stackrox` and became one. Nobody, however, was brave enough yet to change `module github.com/stackrox/rox` to `module github.com/stackrox/stackrox` and address all further renames in all files that import `github.com/stackrox/rox`/anything and do this for https://github.com/stackrox/scanner repo.
Therefore, the code lives in https://github.com/stackrox/stackrox repo but thinks it is `github.com/stackrox/rox` Go module.

Addressing this thing would be a great contribution.

**Comment by RTann** [comment]
The module's path defined in the go.mod does not necessarily need to match the repository. It is just common practice to do so, as it is already a unique name. One example of the module path not matching is when it is decided to bump the major version of the module. For example, StackRox Scanner has the following defined in the [scanner repo](https://github.com/stackrox/scanner/blob/master/go.mod):

```
module github.com/stackrox/scanner
```

however, we are in the process of [migrating the Scanner codebase](https://github.com/stackrox/stackrox/blob/master/scanner/go.mod) and we have bumped the version to v4:

```
module github.com/stackrox/scanner/v4
```

Another example of a major version change is here: https://github.com/go-git/go-git/blob/master/go.mod#L1

An example of a go.mod with a name which does not match any kind of repository: https://github.com/golang/go/blob/master/src/go.mod#L1

See https://go.dev/ref/mod#go-mod-file-ident for more information.

**Comment by RTann** [comment]
I did some more digging, and I found that I was mistaken a bit.

Go does derive the source from the module's path: https://go.dev/ref/mod#vcs-find. So, as long as the required `<meta>` tag exists at whatever endpoint the module's path specifies, the module path does not need to match the source repo.

I'm going to guess that go understands that it does not need to reach out to any VCS for paths which match the module's own path. As for how other repos may be able to depend on the stackrox repo, the scanner repo is an example. It `require`s `github.com/stackrox/rox`, but it replaces it with `github.com/stackrox/stackrox`. See https://github.com/stackrox/scanner/blob/master/go.mod#L161.
